File: apps/windows/DepartmentLauncher.cs
namespace Biupiu.Desktop;

public sealed record DepartmentLaunchResult(
    string Route,
    string PackageName,
    bool Started,
    string Reason);

public static class DepartmentLauncher
{
    public static DepartmentLaunchResult Launch(RuntimeSession session, string route)
    {
        var runtime = DepartmentRuntime.Resolve(session, route);
        return new(
            route,
            runtime.Target.PackageName,
            runtime.CanStart,
            runtime.CanStart ? "STARTED" : runtime.Reason);
    }
}

File: mini-os/android/apps/rnd-os/app/src/main/java/com/biupiu/rndos/DepartmentLauncher.kt
package com.biupiu.rndos

data class DepartmentLaunchResult(
    val route: String,
    val packageName: String,
    val started: Boolean,
    val reason: String
)

object DepartmentLauncher {
    fun launch(session: RuntimeSession, route: String): DepartmentLaunchResult {
        val runtime = DepartmentRuntime.resolve(session, route)
        return DepartmentLaunchResult(
            route,
            runtime.target.packageName,
            runtime.canStart,
            if (runtime.canStart) "STARTED" else runtime.reason
        )
    }
}
